web/src/api/hooks/models.ts

```diff
@@ -49,6 +50,68 @@ export function addModel(data: Model): Promise<Model> {
     })(JSON.stringify(data));
 }
 
+export function deployModel(id: string): Promise<{ [key: string]: string }> {
+    const body = {
+        id
+    };
+    return useBadgerFetch<{ [key: string]: string }>({
+        url: `${namespace}/models/deploy`,
+        method: 'post',
+        withCredentials: true
+    })(JSON.stringify(body));
+}
+
+export function editModel(data: Model): Promise<Model> {
+    const body = { ...data };
+    // TODO: replace this hardcode with real data (update UI)
+    body.description = 'New model';
+    return useBadgerFetch<Model>({
+        url: `${namespace}/models/update`,
+        method: 'put',
+        withCredentials: true
+    })(JSON.stringify(body));
+}
+
+export const useEditModelMutation: MutationHookType<Model, Model> = () => {
+    const queryClient = useQueryClient();
+
+    return useMutation(editModel, {
+        onSuccess: () => {
+            queryClient.invalidateQueries('models');
+        }
+    });
+};
+
+function undeployModel(id: string): Promise<{ [key: string]: string }> {
+    const body = {
+        id
+    };
+    return useBadgerFetch<{ [key: string]: string }>({
+        url: `${namespace}/models/undeploy`,
+        method: 'delete',
+        withCredentials: true
+    })(JSON.stringify(body));
+}
+
+function deleteUndeployedModel(id: string): Promise<{ [key: string]: string }> {
+    const body = {
+        id
+    };
+    return useBadgerFetch<{ [key: string]: string }>({
+        url: `${namespace}/models/delete`,
+        method: 'delete',
+        withCredentials: true
+    })(JSON.stringify(body));
+}
+
+export async function undeployAndDeleteModel(id: string): Promise<void> {
+    try {
+        await undeployModel(id).then(() => deleteUndeployedModel(id));
```

Comment:
I believe we can do just like this here:
await undeployModel(id);
await deleteUndeployedModel(id);